src/ui/views/ViewMain.qml

```diff
@@ -206,8 +207,15 @@ VPNFlickable {
                 }
             ]
 
-            onOpened: VPNSettings.tipsAndTricksIntroShown = true
-            onClosed: tipsAndTricksIntroPopupLoader.active = false
+            onOpened: {
+                VPNSettings.tipsAndTricksIntroShown = true
+                Sample.tipsAndTricksModalShown.record();
+            }
+
+            onClosed: {
```

Comment:
Does this `onClosed` event fire regardless of which button is clicked? If so, and if the purpose of the ping is to track modal _dismissals_, we may want to attach this specifically to the `onClicked` handlers for `tipsAndTricksIntroPopupGoBackButton` and `tipsAndTricksIntroPopupCloseButton`.

Reply:
Right, I was sort of under the impression that "dismissed" meant closed, which, technically all of the buttons do, but I can go ahead and change it so the event is not fired when the primary button is clicked.

Reply:
I've kept the trigger in `onClosed` because there are technically other ways to close the popup besides the "X" and "Go back" CTA (i.e. hitting "Esc", clicking/pressing outside the modal) but I've prevented the event from firing when closed via the primary "Discover now" CTA

Reply:
The goal is analysis that makes sense.

We'll have a count for tipsAndTricksModalShown

For every modal shown there should be a "modal gone" type event. That could be because it was dismissed or maybe people interacted with it and went to the content.

What events can I sum to see the same number of "modal gone" as "modal shown"?

Reply:
@MattLichtenstein posting your note here:

So as it stands right now (after making the change Lesley suggested), there is no combination of metrics that would equate to the "modal shown" count. Are you alluding to adding another metric for primary button clicks. Because that, alongside dismissals, sums up everything the user can do when they are shown the modal.

I didn't add that originally because it was specifically called out in the spec. What I did have originally was that those primary button clicks also triggered a "modal dismissed" event (so dismissed events would equal shown events), but I have since removed that. I could bring that back if you think that is the better option

@lesleyjanenorton I'd like your input here.

Reply:
> Given that this event is called "Sample.tipsAndTricksModalClosed" we could extra key of "how" with "dismissed" | "cta" as values.

I like it. Let's handle it this way.